index/scorch/introducer.go

```diff
@@ -438,8 +436,17 @@ func (s *Scorch) introduceMerge(nextMerge *segmentMerge) {
 		_ = rootPrev.DecRef()
 	}
 
-	// notify requester that we incorporated this
-	nextMerge.notify <- newSnapshot
+	// check for index closure before bumping the reference count,
+	// else this snapshot may leak if the requester has already
+	// exited the work loop.
+	select {
+	case <-s.closeCh:
```

Comment:
hmmm -- a perhaps more robust way might be instead like...
```
newSnapshot.AdRef()
... do stuff
select {
  case <-s.closeCh:
     newSnapshot.DecRef()
  case nextMerge.notify <- newSnapshot:
    // no-op
}
close(nextMerge.notify)
```

And, anytime we respond to the closing of the closeCh, look for these patterns where we still own the ref-count?

Reply:
"more robust" might be ill-worded... perhaps "a little more idiomatic-feeling to me" on how to do this might be the better description

Reply:
yea..I totally agree that this looks more idiomatic. 
Have one doubt here,
iirc => select randomly picks one out of a 'n' valid/successful cases. 
Here : 
"case nextMerge.notify <- newSnapshot: " will always be true as notify is a buffered channel.
So we still have a chance of leaking the bumped index snapshot if the select ends up randomly picking up the second notify ch push case?

Reply:
done!

Reply:
done!